src/opera/pge/dswx_hls/dswx_hls_pge.py

```diff
@@ -74,6 +74,50 @@ def _validate_inputs(self):
 
                 self.logger.critical(self.name, ErrorCode.INVALID_INPUT, error_msg)
 
+    def _validate_expected_input_platforms(self):
+        """
+        Scans the input files to make sure that the data comes from expected
+        platforms only.  Currently, Landsat 8/9, or Sentinel 2 A/B.
+        Raises an exception if an unsupported platform is detected.
+        This function assumes that input files have been checked to exist.
+        It also assumes that only files that contain the metadata keys
+        LANDSAT_PRODUCT_ID for Landsat input, and PRODUCT_URI for Sentinel
+        input, need to be checked.
+        """
+        self.logger.info(
+            self.name, ErrorCode.UPDATING_PRODUCT_METADATA,
+            f'Scanning DSWx input datasets for invalid platforms.'
+        )
+
+        # Get a list of input files to check for invalid platform metadata
+        for input_file in self.runconfig.input_files:
+            input_file_path = abspath(input_file)
+            if isdir(input_file_path):
+                list_of_input_tifs = glob.glob(join(input_file_path, '*.tif'))
+            else:
+                list_of_input_tifs = [input_file_path]
```

Comment:
I think this case needs to be appending `input_file_path` to `list_of_input_tifs`. This is because inputs could be specified in one of two ways: as a directory containing the 7 or so input HLS bands (covered by your `isdir` case), and as separately enumerated paths to each input HLS band. As-is, this would overwrite `list_of_input_tifs` for each `input_file`.